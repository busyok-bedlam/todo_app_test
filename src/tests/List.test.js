import React from 'react';
import List from '../components/list';

const todoes = [
    {_id: 5, title: "some title", body: "dsklhfdjksnvmc", done: true},
    {_id: 6, title: "title", body: "xxxxxxxx", done: false}

]
jest.mock('../containers/form-page', () => ()=> <div id="mockUserCom">KeyFigures</div>)
jest.mock('../containers/item-page', () => ()=> <div >List item</div>)

describe('<List />', () => {
    it('renders correctly', () => {
        const didMountSpy = sinon.spy(List.prototype, 'componentDidMount');
        const wrapper = mount(<List todos={todoes} />);
        expect(didMountSpy.callCount).toBe(1);
        expect(wrapper).toMatchSnapshot()
    })
})