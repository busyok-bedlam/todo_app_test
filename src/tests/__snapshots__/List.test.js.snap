// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders correctly 1`] = `
<List
  todos={
    Array [
      Object {
        "_id": 5,
        "body": "dsklhfdjksnvmc",
        "done": true,
        "title": "some title",
      },
      Object {
        "_id": 6,
        "body": "xxxxxxxx",
        "done": false,
        "title": "title",
      },
    ]
  }
>
  <div
    style={
      Object {
        "height": "50vh",
        "overflowY": "scroll",
      }
    }
  >
    <Component
      data={
        Object {
          "_id": 5,
          "body": "dsklhfdjksnvmc",
          "done": true,
          "title": "some title",
        }
      }
      key="0"
    >
      <div>
        List item
      </div>
    </Component>
    <Component
      data={
        Object {
          "_id": 6,
          "body": "xxxxxxxx",
          "done": false,
          "title": "title",
        }
      }
      key="1"
    >
      <div>
        List item
      </div>
    </Component>
  </div>
</List>
`;
